### PR TITLE
Optional Reply Queue for Exchanges

### PR DESCRIPTION
**Pull request by Akamaozu**
First pass at fixing this issue: https://github.com/pagerinc/jackrabbit/issues/69

Note
---
**Made a conscious decision to skip checking for replyQueue-related things in the RPC bit of the code**

Completely unfamiliar with the usecase, so I don't know if reply queues can be optional there. Would appreciate some clarity on how best to address the `noReply`-RPC situation.

### CODE REVIEW

**Comment by cilindrox** [comment]
Thanks @Akamaozu we'll take a look at this.

cc @ryanhallcs

File: lib/exchange.js [hunk 1]
```diff
@@ -320,18 +330,25 @@ const exchange = (name, type, options) => {
             return bail(err);
         }
 
-        replyQueue.connect(connection);
-        replyQueue.once('ready', () => {
-
+        if (!replyQueue) {
             ready = true;
             emitter.emit('ready');
-        });
```
Comment (cilindrox) [inline lib/exchange.js]: do we want to `return`/break here instead of falling through the `else` block?

Comment (Akamaozu) [reply lib/exchange.js]: Since they'll behave exactly the same way on this situation, I'm happy with whichever strategy team @pagerinc prefers.

Comment (ryanhallcs) [reply lib/exchange.js]: it's a nitpick, but we prefer `return`s over `else` branches

**Comment by ryanhallcs** [comment]
LGTM, just a small style note, thank you for this!

**Comment by cilindrox** [comment]
Cutting v4.8.0 after we merge this

File: lib/exchange.js (hunk 1 above)
Comment (Akamaozu) [reply lib/exchange.js]: consider your nits picked, sirs.

**Comment by cilindrox** [comment]
cut v4.8.0 - thanks @Akamaozu !

**Pull request merged**